Acornfile

```diff
@@ -5,100 +5,127 @@ info:        localData.info
 icon:        "./icon.png"
 
 args: {
-  // Name of the default database
-  dbName: "mydb"
+	// Name of the default database
+	dbName: "mydb"
 
-  // Name of the default user
-  dbUser: ""
+	// Name of the default user
+	dbUser: ""
+
+	// Number of MongoDB instances to run in the MongoDB replica set
+	replicas: number > 0 || default 1
 }
 
 services: db: {
 	default:   true
-	container: "mongodb"
+	container: "mongodb-0"
 	secrets: ["admin", "user"]
 	ports: "27017"
 	data: dbName: args.dbName
 }
 
 containers: {
-  mongodb: {
-    name: "MongoDB"
-    description: "Container running a MongoDB server"
-    image: "mongo:6.0"
-    ports: "27017/tcp"
-    env: {
-      MONGO_INITDB_ROOT_USERNAME: "secret://admin/username"
-      MONGO_INITDB_ROOT_PASSWORD: "secret://admin/password"
-    }
-    dirs: {
-      "/data/db": "volume://db-data"
-    }
-    probes: [
-      {
-        type:                "liveness"
-        initialDelaySeconds: 10
-        timeoutSeconds:      5
-        tcp:{
-          url: "tcp://localhost:27017"
-        }
-      },   
-    ]
-  }
+	for i in std.range(0, args.replicas) {
+		"mongodb-\(i)": {
+			if i == 0 {
+				sidecars: "init-replica-set-key": {
+					init:  true
+					image: "mongo:6.0"
+					// Replica set keys must be between 6 and 1024 characters in length and consist of only valid base64 characters.
+					// See https://www.mongodb.com/docs/v6.2/tutorial/deploy-replica-set-with-keyfile-access-control/#create-a-keyfile
+					// for details.
+					entrypoint: ["bash", "-c", "openssl rand -base64 756 > replica-set-key && chmod 400 replica-set-key && chown 999:999 replica-set-key && mv replica-set-key /var/keys/replica-set-key"]
```

Comment:
If possible think this sidecar should be simplified to a generated secret that gets mounted as a file w/ the correct permissions (the default permissions for `files` is too permissive for mongo, and it complains)